core/templates/dev/head/pages/email_dashboard/EmailDashboardResult.js

```diff
@@ -17,21 +17,19 @@
  */
 
 oppia.controller('EmailDashboardResult', [
-  '$scope', '$http', '$window', '$timeout',
-  function($scope, $http, $window, $timeout) {
-    var RESULT_HANDLER_URL_PREFIX = '/emaildashboardresult';
-    var CANCEL_EMAIL_HANDLER_URL_PREFIX = '/emaildashboardcancelresult';
+  '$scope', '$http', '$window', '$timeout', 'UrlInterpolationService',
+  function($scope, $http, $window, $timeout, UrlInterpolationService) {
+    var RESULT_HANDLER_URL = '/emaildashboardresult/<query_id>';
+    var CANCEL_EMAIL_HANDLER_URL = '/emaildashboardcancelresult/<query_id>';
     var EMAIL_DASHBOARD_PAGE = '/emaildashboard';
-    var TEST_BULK_EMAIL_URL_PREFIX = '/emaildashboardtestbulkemailhandler';
+    var TEST_BULK_EMAIL_URL = '/emaildashboardtestbulkemailhandler/<query_id>';
 
     var getQueryId = function() {
       return $window.location.pathname.split('/').slice(-1)[0];
     };
 
-    $scope.submitEmail = function() {
-      var resultHandlerUrl = RESULT_HANDLER_URL_PREFIX + '/' + getQueryId();
+    var validateEmailSubjectAndBody = function() {
```

Comment:
It's actually quite confusing for a return value of true to indicate that the data is *invalid*. Can you flip it so that this returns true iff the data is valid?